packages/ckb-indexer/src/transactionCollector.ts

```diff
@@ -247,49 +248,37 @@ export class CKBTransactionCollector extends BaseIndexerModule.TransactionCollec
   };
 
   private isCellScriptArgsValidate = (targetCell: Output) => {
-    let lockArgsLen: number | "any" | undefined;
-    let typeArgsLen: number | "any" | undefined;
     if (this.queries.lock) {
-      lockArgsLen = instanceOfScriptWrapper(this.queries.lock)
+      let lockArgsLen = instanceOfScriptWrapper(this.queries.lock)
         ? this.queries.lock.argsLen
         : this.queries.argsLen;
+      if (lockArgsLen && lockArgsLen !== -1 && lockArgsLen !== 'any' && getHexStringBytes(targetCell.lock.args) !== lockArgsLen) {
```

Comment:
early return is a good pattern for writing more readable code, maybe we can refactor code like this

```ts

private isLockArgsLenMatched = (args: string, argsLen?: ArgsLen) => {
  if (!argsLen) return true;
  if (argsLen === 'any') return true;
  ...
}

private isTypeArgsLenMatched = (args: string, argsLen: ArgsLen ) => {
  ...
}

private isCellScriptArgsMatched = (cell: Output) => {
  const lockArgsLen = instanceOfScriptWrapper(this.queries.lock) ? this.queries.lock.argsLen : this.queries.argsLen;
  const typeArgsLen = instanceOfScriptWrapper(this.queries.type) ? this.queries.type.argLen: this.queries.argsLen;

  return this.isLockArgsLenMatched(cell.lock.args, lockArgsLen) && this.isTypeArgsLenMatched(typeArgs, typeArgsLen)
}

```

Reply:
refactor(ckb-indexer): early return

Reply:
refactor(ckb-indexer): early return